File: src/main/java/com/sequenceiq/cloudbreak/repository/ClusterRepository.java
package com.sequenceiq.cloudbreak.repository;

import java.util.Set;

import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.security.access.prepost.PostAuthorize;

import com.sequenceiq.cloudbreak.domain.Cluster;

public interface ClusterRepository extends CrudRepository<Cluster, Long> {

    @PostAuthorize("returnObject?.owner == principal")
    Cluster findOne(@Param("id") Long id);

    Cluster findById(@Param("id") Long id);

    Set<Cluster> findAllClusterByBlueprint(@Param("id") Long id);

    Cluster findOneWithLists(@Param("id") Long id);

}

File: autoscale/src/main/java/com/sequenceiq/periscope/repository/ClusterRepository.kt
package com.sequenceiq.periscope.repository

import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param
import org.springframework.security.access.prepost.PostAuthorize

import com.sequenceiq.periscope.domain.Cluster
import com.sequenceiq.periscope.api.model.ClusterState

interface ClusterRepository : CrudRepository<Cluster, Long> {

    @PostAuthorize("hasPermission(returnObject,'read')")
    override fun findOne(@Param("id") id: Long?): Cluster

    fun find(@Param("id") id: Long?): Cluster

    fun findAllByUser(@Param("id") id: String): List<Cluster>

    fun findAllByState(@Param("state") state: ClusterState): List<Cluster>
}
